### PR TITLE
Actions with parameters

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR makes McJava use parameters of rendering actions recently [introduced in ProtoData](https://github.com/SpineEventEngine/ProtoData/pull/257).

## Chages in details
 * `SuperInterface` proto type was introduced in order to allow passing generic types with their arguments.
 * Render action classes like `ImplementUuidValue` and `ImplementEventMessage` were eliminated because we now can use `ImplementInterface` action with the `SuperInterface` parameter.
 * `WithActionList` interface was renamed to `WithActions` with appropriate change of getter signature.
 * `io.spine.tools.mc.java.field.FieldClass` was renamed to `AddFieldClass` and is now used with the `StringValue` paramter.
 * Proto properties related to settings that are no longer used were removed.

## Other notable changes

 * **Settings classes for Gradle were converted to Kotlin**
   Previously, part of classes in the `io.spine.tools.mc.java.gradle.settings` were in Java and newer ones were in Kotlin. Since Kotlin is the preferred language for Gradle, there's no much sense in having implementation in Java.

 * **Fixed parameter/property descriptions** according to recently [updated conventions](https://github.com/SpineEventEngine/documentation/wiki/KDoc-and-Dokka#documenting-a-type).
 
 * **The parameter called `p` passed to constructors of action classes was renamed to `project`** for disambiguation with the action parameter.

 * **Fixed package for Gradle settings classes** that were previously misplaced.
 
 * Fix: **Only top-level messages are included into grouping**.  Previously, nested messages were also included into code generation.

### CODE REVIEW

File: mc-java-base/src/main/kotlin/io/spine/tools/mc/java/ImplementInterface.kt [hunk 1]
```diff
@@ -30,32 +30,31 @@ import io.spine.protodata.CodegenContext
...
  *
- * @property type the type of the message.
- * @property file the source code to which the action is applied.
- * @property superInterface the interface to implement.
- * @property genericParameters optional generic parameters if [superInterface] is a generic type.
- * @property context the code generation context in which this action runs.
+ * @param type The type of the message.
+ * @param file The source code to which the action is applied.
+ * @param superInterface The interface to implement.
+ * @param context the code generation context in which this action runs.
```
Comment (armiol) [inline mc-java-base/src/main/kotlin/io/spine/tools/mc/java/ImplementInterface.kt:45]: "The".

File: mc-java-base/src/main/kotlin/io/spine/tools/mc/java/TuneNestedClass.kt [hunk 2]
```diff
@@ -35,17 +37,17 @@ import io.spine.tools.code.Java
 /**
  * Abstract base for code generators modifying classes nested under Java classes of message types.
  *
- * @property type the type of the message.
- * @property file the source code to which the action is applied.
+ * @param type the type of the message.
+ * @param file the source code to which the action is applied.
  * @property simpleName a simple name of the nested class to be modified.
- * @property context the code generation context in which this action runs.
+ * @param context the code generation context in which this action runs.
```
Comment (armiol) [inline mc-java-base/src/main/kotlin/io/spine/tools/mc/java/TuneNestedClass.kt:43]: "The".

File: mc-java-base/src/main/kotlin/io/spine/tools/mc/java/field/AddFieldClass.kt [hunk 3]
```diff
@@ -35,27 +36,29 @@ import io.spine.protodata.java.ClassName
...
 import io.spine.tools.psi.java.addLast
 import org.intellij.lang.annotations.Language
 
 /**
  * Creates a nested class called [`Field`][NAME] under a Java class generated for
  * the given message [type].
  *
- * @property type the message type for the Java code of which to generate the nested class.
- * @property file the file with Java code generated for the [type].
+ * @param type the message type for the Java code of which to generate the nested class.
```
Comment (armiol) [inline mc-java-base/src/main/kotlin/io/spine/tools/mc/java/field/AddFieldClass.kt:47]: "The".

Same below.

File: mc-java-base/src/main/kotlin/io/spine/tools/mc/java/gradle/settings/SettingsWithActions.kt [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        actions.put(className, stringValue.pack())
+    }
+
+    /**
+     * Instructs Model Compiler to apply
+     * [code generation actions][io.spine.protodata.renderer.RenderAction]
+     * to the code generated for messages of this group.
+     *
+     * @param classNames
+     * the binary names of the action class
```
Comment (armiol) [inline mc-java-base/src/main/kotlin/io/spine/tools/mc/java/gradle/settings/SettingsWithActions.kt:158]: Let's move this line one line back, and start with "The".

File: mc-java-base/src/main/kotlin/io/spine/tools/mc/java/gradle/settings/SettingsWithActions.kt [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        useAction(
+            ImplementInterface::class.java.name,
+            superInterface { name = interfaceName }
+        )
+    }
+
+    /**
+     * Configures the associated messages to implement a Java interface with the given name.
+     *
+     * The declaration of the interface in Java must exist. It will not be generated.
```
Comment (armiol) [inline mc-java-base/src/main/kotlin/io/spine/tools/mc/java/gradle/settings/SettingsWithActions.kt:92]: As long as I recall, previously, we were generating a new marker interface in this case.

At the same time, I am not sure why we did that, and also, I don't know if anyone has used such a feature. So there may be no consequences. But still, let's discuss the matter to understand whether we should do anything after this PR is merged.

File: mc-java-base/src/main/proto/spine/tools/mc/java/settings.proto [hunk 6]
```diff
@@ -248,7 +241,21 @@ message Uuids {
...
+message SuperInterface {
+
+    // The name of the interface.
+    //
+    // It could be a simple name if the generated types are in the same Java package.
+    // Otherwise, this value must contain a fully-qualified name.
+    //
+    string name = 1 [(required) = true];
+
+    // Optional generic parameters.
```
Comment (armiol) [inline mc-java-base/src/main/proto/spine/tools/mc/java/settings.proto:257]: Are these just letters (e.g. `S`, as one may think)? Or are these the FQNs? Please specify.

File: mc-java-entity/src/main/kotlin/io/spine/tools/mc/java/entity/ImplementEntityState.kt [hunk 7]
```diff
@@ -26,53 +26,57 @@
...
  *
  * The class is public because its fully qualified name is used as a default
- * value in [UuidSettings][io.spine.tools.mc.java.gradle.settings.EntitySettings].
+ * value in [EntitySettings][io.spine.tools.mc.java.gradle.settings.EntitySettings].
  *
- * @property type the type of the message.
- * @property file the source code to which the action is applied.
- * @property context the code generation context in which this action runs.
+ * ## Implementation note
+ * The class descends from [DirectMessageAction] and delegates to [ImplementInterface] in
```
Comment (armiol) [inline mc-java-entity/src/main/kotlin/io/spine/tools/mc/java/entity/ImplementEntityState.kt:50]: Please allow an empty line above this one.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov mostly LGTM, please see my comments, though.

File: mc-java-base/src/main/kotlin/io/spine/tools/mc/java/gradle/settings/SettingsWithActions.kt (hunk 5 above)
Comment (alexander-yevsyukov) [reply mc-java-base/src/main/kotlin/io/spine/tools/mc/java/gradle/settings/SettingsWithActions.kt:92]: It's the way we used to have. We have the `generate` field in `IsOption` type which regulates the generation of Java interfaces. But we always required the presence of an interface specified via Gradle project settings.

**Comment by alexander-yevsyukov** [comment]
@yevhenii-nadtochii, have a look at the changes introduced by this PR. It might affect your current works, but, hopefully, not much.

**Pull request merged**